pages/index.tsx

```diff
@@ -1,156 +1,206 @@
-      name: receiverName,
-      gift: giftName,
-      keyID: generateKeyID(),
-    };
+  function handleDeletion(gift: FullLocalStorage) {
+    const confirmDeletion = confirm(`Deleting ${gift.name} - ${gift.gift}`);
+    if (confirmDeletion) {
+      let localStorageGifts: FullLocalStorage[] = JSON.parse(
+        getLocalStorage('giftData'),
+      );
+      localStorageGifts = localStorageGifts.filter(
+        (localStorageGift) => localStorageGift.id !== gift.id,
+      );
+      setLocalStorage('giftData', JSON.stringify(localStorageGifts));
+      refreshGiftList();
+    }
+  }
 
-    setLocalStorage(generateID(), JSON.stringify(JSON_Object));
-    setGiftData((previousValue) => previousValue.concat(JSON_Object));
+  function refreshGiftList() {
+    const sortedGifts = sortGiftsOldestFirst(
+      JSON.parse(getLocalStorage('giftData')),
+    );
+    setGiftData(sortedGifts);
   }
 
   return (
-    <main className={`bg-white w-full max-w-full h-screen ${inter.className}`}>
-      <div id="fullScreenContainer" className="justify-center grid h-5">
-        <div id="formContainer" className="mt-5">
-          <form id="giftForm" className="" action={'/api/create'}>
-            <div id="formTitle" className="text-2xl pt-4">
-              Lahjalistaidea
-            </div>
-
-            <div id="giftNameContainer" className="pt-4 grid">
-              <label htmlFor="giftName">Lahja</label>
-              <input
+    <Main className={`bg-white w-full max-w-full h-screen ${inter.className}`}>
+      <Container className="justify-center grid h-5">
+        <Container className="mt-5">
+          <Form onSubmit={(e) => e.preventDefault()}>
+            <TitleText className="text-2xl pt-4">Lahjalistaidea</TitleText>
+            <Container className="pt-4 grid">
+              <Label htmlFor="giftName">Lahja</Label>
+              <Input
                 id="giftName"
+                required={true}
+                onChange={(event) => handleGiftNameChange(event)}
+                onInvalid={(event) =>
+                  (event.target as HTMLInputElement).setCustomValidity(' ')
+                }
+                autoComplete="off"
                 type="text"
                 className="ps-1 pt-3 pb-3 border hover:bg-gray-100"
                 placeholder="Kortti"
                 name="giftName"
+                value={newGiftName}
               />
-            </div>
-            <div id="giftReceiverContainer" className="pt-4 grid">
-              <label htmlFor="receiver">Saaja</label>
-              <input
+              {giftNameError && (
+                <div className="text-red-500">Lahja on pakollinen</div>
+              )}
+            </Container>
+            <Container className="pt-4 grid">
+              <Label htmlFor="receiver">Saaja</Label>
+              <Input
                 id="giftReceiver"
+                required={true}
+                onChange={(event) => handleReceiverChange(event)}
```

Comment:
Yksinkertaisempi laittaa inline tähän
```suggestion
                onChange={(event) => setNewReceiver(event.target.value)}
```